### DISCUSSION TITLE
Releases and Packages

### DISCUSSION THREAD

**Discussion by ghost** [astarub/campus_app#51] [Ideas]
1. I would suggest to use the Release feature https://github.com/astarub/campus_app/releases as it makes it easier to keep track of the progress towards a new version and the planned features.
2. It would be nice to have a downloadable APK File with those releases to install the App without the need for Play or Appstore.
3. In the long run it would be awesome to see the App being included in F-Droid.

**Comment by C4s4r** [comment]
Point 1 and 2 go hand-in-hand, as we want to include APK-files in the GitHub releases.
If I remember correctly, @domai-tb was planning to work on the integration of GitHub releases in our workflow.

F-Droid releases are on our internal backlog, but this is currently handled with low priority.

**Comment by domai-tb** [comment]
Yes, im allready worked on a release workflow for GitHub and it is nearly done. Currently we use Codemagic as CI/CD tool to manage our release to the official Google and Apple Store, which would lead to inconsistency in the version numbers. Codemagic pulls the current release number from the store and automatically increase the build number, so the GitHub workflow should do the same. 

The current workflow of PR #42 is just using the current date to name the APK-files. So we could get the current build number by setup the [Codemagic Google Play CLI-Tool](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/google-play/README.md) inside the workflow (or changing our current CI/CI-pipeline, what is currently unplaned). But feel free to add your input!

A F-Droid release is also planned but theire we are facing similiar issues. As mentioned above, Codemagic is managing our build number. So first we have to get ot from their and than publishing this repository to the official [F-Droid Store](https://gitlab.com/fdroid/fdroiddata) by doing a pull request. We didn't find a straigth-forward way to do so, so thats the reason why their is no F-Droid release yet. 

Cheers, :Domai

**Reply by Nicole4Fox** [reply]
Anything new on this?

**Reply by domai-tb** [reply]
Unfortunately, no one has continued to work on it so far.
